posts/navarroaxel/20.git-rebase-edit.md

```diff
@@ -0,0 +1,58 @@
+---
+title: A step-by-step guide to splitting commits using Git rebase
+published: true
+description: Learn why and how to split commits to keep your Git history clean and organized. Whether you're backporting hotfixes or simply aiming for more readable commit history, this step-by-step guide has got you covered.
+tags: 'git, bash, productivity, tooling'
+cover_image: ./assets/git-rebase-edit.png
+id: 2015782
+---
+
+In the previous posts we checked how to use the [`git rebase --interactive`](https://git-scm.com/docs/git-rebase#Documentation/git-rebase.txt---interactive) and how to squash your commits using the `fixup` command in the `git-rebase-todo` file.
+
+Now, we're going to learn how to split commits using the terminal. But why split a commit? This helps organize your work. For instance, you may need to rearrange your changes to make backporting hotfixes to older released versions easier, likely using `git cherry-pick`. Let’s dive in!
+
+## Splitting commits
+
+You can use `git rebase` to split your commits. There are several alternatives to do it and we're going to check the `git rebase --interactive` way. You can use `git rebase --interactive main` to reapply all your new commits to the top of the `main` branch, or just `git rebase --interactive HEAD~2` to apply the latest 2 commits.
+
+```text
+pick 0680aa7 fix header responsiveness
+exec npm test
+edit 68ab202 fix grid responsiveness and upgrade bootstrap
+exec npm test
+break
+```
+
+💡 You can `pick` the rebased commits in any order! but this can cause Git conflicts 😞. Also you can `pick` commits from other branches like [`git cherry-pick`](https://git-scm.com/docs/git-cherry-pick).
+
+Well, Git will `pick` the commit `0680aa7` as-is and run `npm test` and if the command exists successfully then Git will pick the commit `68ab202` and break returning the control of the shell to you allowing to `edit` this last commit. Let's see how to edit the last commit.
+
+🧠 To clarify: `edit` stops after applying the commit, so that you can edit the files or the commit message, amend the commit, and continue rebasing.
+
+First, undo it keeping all the changes in the working tree using the following command:
+
+```bash
+git reset HEAD^
+```
+
+Now, you are able to make any edition to your code. Then you can use `git add` to add the changes to the index to commit it in parts splitting the original commit. As the following example shows:
+
+```bash
+git add package.json package-lock.json
+git commit -m 'upgrade bootstrap'
+git add .
+git commit -m 'fix grid responsiveness'
+git rebase --continue
+```
+
+Git will continue running the second `exec` command that makes an `npm test`, if this fails you need to fix your code, otherwise Git will continue with the `break` command returning the control of the shell to you again. Here you are able to run your application, make the final checks, etc.
```

Comment:
```suggestion
Git will continue running the second `exec` command that makes an `npm test`. If this fails you will need to fix your code; otherwise Git will continue with the `break` command returning the control of the shell to you again. Here you are able to run your application, make the final checks, etc.
```

"Continue with the `break` command" sounds weird